### ISSUE TITLE
PROJECT TEST CASE 9_11

### ISSUE THREAD

**Issue by gerardgoh** [nus-cs2030/2324-s2#408]
## Description 
Hi not sure what went wrong after line 33 and the differences in timing. Hope someone can explain how it went wrong

Thanks!

## Screenshots (if any): 

<img width="612" alt="Screenshot 2024-04-01 at 4 57 39 PM" src="https://github.com/nus-cs2030/2324-s2/assets/132872750/c3a85ff6-3906-4ec3-9e95-85a6dc3f47e9">

**Comment by jiancheng37** [comment]
when a customer is done being served by a self checkout, the next customer in queue should be served by that self checkout. do check your priority queue and see why this isnt happening

**Comment by zuoshihua** [comment]
You may find my issue https://github.com/nus-cs2030/2324-s2/issues/361 useful

tl;dr your customer should check if other self checkouts are available. If all are busy, customer should wait for the soonest available self checkout

**Comment by Jeremythong2002** [comment]
Yep, depending on your implementation, every time a self-check out finishes, it should check the next customer in the shared queue, so customers are not tied to a particular self-check.

Importantly, check that your Wait event for the other customers that are waiting for the self-check is updated whenever you carry out this operation.

**Comment by tianyiiiiiii** [comment]
I had a similar problem too, you can try to loop through all the self-checkouts to find out which one is done the earliest while the customer is waiting

**Comment by ZHD1987E** [comment]
Did you manage to solve it?

If yes, please close the issue yeah?
